GREYErrorFormatter.m

```diff
@@ -0,0 +1,195 @@
+                                          kErrorFormatterSearchActionInfoKey,
+                                          kErrorFormatterScreenshotPathsKey,
+                                          kErrorFormatterUnderlyingErrorKey,
+                                          kErrorFormatterUIHierarchyKey,
+                                          kErrorFormatterStackTraceKey]];
+  }
+  GREYFatalAssertWithMessage(false, @"Error Domain and Code Not Yet Supported");
+}
+
+- (NSString *)loggerDescriptionForKeys:(NSSet<NSString *> *)keys {
+  NSMutableArray<NSString *> *logger = [[NSMutableArray alloc] init];
+  
+  if ([keys containsObject:kErrorFormatterExceptionReasonKey]) {
+    NSString *exceptionReason = _error.localizedDescription;
+    [logger addObject:[NSString stringWithFormat:@"\n%@\n", exceptionReason]];
+  }
+  
+  if ([keys containsObject:kErrorFormatterRecoverySuggestionKey]) {
+    NSString *recoverySuggestion = _error.userInfo[kErrorDetailRecoverySuggestionKey];
+    if (recoverySuggestion) {
+      [logger addObject:[NSString stringWithFormat:@"%@\n", recoverySuggestion]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterElementMatcherKey]) {
+    NSString *elementMatcher = _error.userInfo[kErrorDetailElementMatcherKey];
+    if (elementMatcher) {
+      [logger addObject:[NSString stringWithFormat:@"%@:\n%@\n", kErrorDetailElementMatcherKey,
+                         elementMatcher]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterSearchActionInfoKey]) {
+    NSString *searchActionInfo = _error.userInfo[kErrorDetailSearchActionInfoKey];
+    if (searchActionInfo) {
+      [logger addObject:[NSString stringWithFormat:@"%@\n%@\n", kErrorDetailSearchActionInfoKey,
+                         searchActionInfo]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterScreenshotPathsKey]) {
+    for (NSString *key in _error.appScreenshots.allKeys) {
+      NSString *screenshotPath = _error.appScreenshots[key];
+      [logger addObject:[NSString stringWithFormat:@"%@: %@\n", key, screenshotPath]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterUnderlyingErrorKey]) {
+    NSString *nestedError = _error.nestedError.description;
+    if (nestedError) {
+      [logger addObject:[NSString stringWithFormat:@"%@: \n%@\n", kErrorFormatterUnderlyingErrorKey,
+                         nestedError]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterUIHierarchyKey]) {
+    NSString *UIHierarchy = GREYFormattedHierarchy(_error.appUIHierarchy);
+    if (UIHierarchy) {
+      [logger addObject:UIHierarchy];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterStackTraceKey]) {
+    NSArray<NSString *> *stackTrace = _error.stackTrace;
+    if (stackTrace) {
+      [logger addObject:[NSString stringWithFormat:@"%@: %@\n", kErrorStackTraceKey, stackTrace]];
+    }
+  }
+  
+  return [logger componentsJoinedByString:@"\n"];
+}
+
+#pragma mark - Static Functions
```

Comment:
Static function in C means it's scoped to this file. This function is public right?